monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/BalArtifactsManager.java

```diff
@@ -0,0 +1,96 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.ei.dashboard.micro.integrator;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.ei.dashboard.core.commons.utils.HttpUtils;
+import org.wso2.ei.dashboard.core.data.manager.DataManager;
+import org.wso2.ei.dashboard.core.data.manager.DataManagerSingleton;
+import org.wso2.ei.dashboard.core.exception.ManagementApiException;
+import org.wso2.ei.dashboard.core.rest.delegates.ArtifactsManager;
+import org.wso2.ei.dashboard.core.rest.delegates.heartbeat.HeartbeatObject;
+import org.wso2.ei.dashboard.micro.integrator.commons.Utils;
+import org.wso2.micro.integrator.dashboard.utils.ExecutorServiceHolder;
+
+import java.io.IOException;
+
+/**
+ * Fetch, store, update and delete artifact information of registered Ballerina nodes.
+ */
+public class BalArtifactsManager implements ArtifactsManager {
+
+    private static final Logger logger = LogManager.getLogger(BalArtifactsManager.class);
+    private final DataManager dataManager = DataManagerSingleton.getDataManager();
+    private HeartbeatObject heartbeat = null;
+
+    public BalArtifactsManager(HeartbeatObject heartbeat) {
+        this.heartbeat = heartbeat;
+    }
+
+    @Override
+    public void runFetchAllExecutorService() {
+        Runnable runnable = () -> {
+            String nodeId = heartbeat.getNodeId();
+            String groupId = heartbeat.getGroupId();
+            logger.info("Fetching artifacts from node " + nodeId + " in group " + groupId);
+            String accessToken = dataManager.getAccessToken(groupId, nodeId);
+            try {
+                fetchAndStoreServers(accessToken);
+            } catch (ManagementApiException e) {
+                logger.error("Unable to fetch artifacts/details from node: {} of group: {} due to {} ", nodeId,
+                        groupId, e.getMessage(), e);
+            }
+        };
+        ExecutorServiceHolder.getBalArtifactsManagerExecutorService().execute(runnable);
+    }
+
+    @Override
+    public void runDeleteAllExecutorService() {
+        Runnable runnable = this::deleteAllArtifacts;
+        ExecutorServiceHolder.getBalArtifactsManagerExecutorService().execute(runnable);
+    }
+
+    private void fetchAndStoreServers(String accessToken) throws ManagementApiException {
+        // This code need to be changed according to the Ballerina server API
+        String url = heartbeat.getMgtApiUrl();
```

Comment:
With this URL we can only get nodeId and nodeData (bal-version, bal-home, os and os-version) right? We need to have another URL to get the artifact types (e.g., `String url = heartbeat.getMgtApiUrl() + "SERVICE";`)

Reply:
yes. This is only for server information.

Reply:
We might need to modify this further.